modules/account-map/modules/iam-roles/outputs.tf

```diff
@@ -8,7 +8,7 @@ output "terraform_profile_name" {
 
 output "org_role_arn" {
   value = local.account_name == module.account_map.outputs.root_account_account_name ? null : format(
-    "arn:aws:iam::%s:role/OrganizationAccountAccessRole",
+    "${var.arn_format}:iam::%s:role/OrganizationAccountAccessRole",
```

Comment:
@mikedizon  thanks for the PR.

Instead of the new variable, please use:

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/partition

similar to:
https://github.com/cloudposse/terraform-aws-ecr/blob/58ca80423fc03e97846fbba44efb6aca9b4a3d6b/main.tf#L90

https://truss.works/blog/adventures-in-govcloud

Reply:
@aknysh tried this approach last night. there seems to be a dependency cycle issue when using the `aws_partition` data source. the aws provider in the parent module references the output being modified in the `iam_roles` submodule. 

https://github.com/cloudposse/terraform-aws-components/blob/master/modules/account-settings/providers.tf#L7

adding an aws provider in the `iam_roles` submodule itself fixes this but we'd have to pass a variable for the region.